src/query/service/tests/it/sql/planner/optimizer/eliminate_in_subquery_test.rs

```diff
@@ -0,0 +1,421 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use databend_common_ast::ast::Engine;
+use databend_common_ast::parser::parse_sql;
+use databend_common_ast::parser::tokenize_sql;
+use databend_common_ast::parser::Dialect;
+use databend_common_catalog::catalog::CatalogManager;
+use databend_common_catalog::table_context::TableContext;
+use databend_common_exception::Result;
+use databend_common_expression::types::NumberDataType;
+use databend_common_expression::TableDataType;
+use databend_common_expression::TableField;
+use databend_common_expression::TableSchemaRefExt;
+use databend_common_meta_app::schema::CreateOption;
+use databend_common_sql::optimizer::SExpr;
+use databend_common_sql::optimizer::SubqueryRewriter;
+use databend_common_sql::plans::CreateTablePlan;
+use databend_common_sql::plans::Plan;
+use databend_common_sql::Binder;
+use databend_common_sql::Metadata;
+use databend_common_sql::MetadataRef;
+use databend_common_sql::NameResolutionContext;
+use databend_query::interpreters::CreateTableInterpreter;
+use databend_query::interpreters::Interpreter;
+use databend_query::test_kits::TestFixture;
+use databend_storages_common_table_meta::table::OPT_KEY_DATABASE_ID;
+use databend_storages_common_table_meta::table::OPT_KEY_STORAGE_FORMAT;
+use parking_lot::RwLock;
+
+struct TestSuite {
+    name: &'static str,
+    // Test cases
+    query: &'static str,
+    // Expected results
+    explain: &'static str,
+}
+
+fn get_test_suites() -> Vec<TestSuite> {
+    vec![
+        // Elimination Successful
+        TestSuite {
+            name: "base",
+            query: "SELECT t3.* FROM t1 t3 WHERE t3.b IN (SELECT t4.b FROM t1 t4 WHERE t4.a = 5);",
+            explain: "EvalScalar
+├── scalars: [t3.a (#0) AS (#0), t3.b (#1) AS (#1), t3.c (#2) AS (#2)]
```

Comment:
those tests seems like explain test, we can write them in this file
`tests/sqllogictests/suites/mode/standalone/explain/subquery.test`
here we can use `Matcher` to check result plan after optimize